services/FeefoReviewService.php

```diff
@@ -1,26 +1,24 @@
 <?php
 namespace Craft;
 
-class FeefoReviewService extends BaseApplicationComponent
-{
+class FeefoReviewService extends BaseApplicationComponent {
   public $plugin = null;
   public $settings = array();
 
-  public function __construct()
-  {
+  public function __construct(){
     $this->plugin   = craft()->plugins->getPlugin('feeforeview');
     $this->settings = $this->plugin->getSettings();
   }
 
-  public function getReviewSummary($id)
-  {
-    if ($this->feefoMerchantId() == null) return null;
-
+  public function getReviewSummary($id){
+    $merchantId = $this->feefoMerchantId();
```

Comment:
You've changed this code which has removed an important code paradigm called a "guard". Guards are designed to guard the entry into a function and return early if the conditions are not met. This generally has the affect of simplifying the function by not having it's entire contents wrapped in a condition.

Secondly, you have defined a new variable in place of using a class attribute. This additional definition is not required because the class attribute is easily accessible by calling `$this->feefoMerchantId()`.

You can achieve the result you were after, and keep the existing guard like so:
```
public function getReviewSummary($id){
  if (is_null($this->feefoMerchantId())) return null;
  if (is_null($this->getProducts())) return null;

  $index = array_search($id, array_column($products, 'vendor_ref'));
  return $index !== false ? $products[$index] : null;
}
```